src/into_dart.rs

```diff
@@ -253,12 +257,12 @@ macro_rules! dart_typed_data_type_trait_impl {
             #[doc(hidden)]
             #[no_mangle]
             pub(crate) unsafe extern "C" fn $free_zero_copy_buffer_func(
-                isolate_callback_data: *mut c_void,
+                _isolate_callback_data: *mut c_void,
                 peer: *mut c_void,
             ) {
-                let len = (isolate_callback_data as isize) as usize;
-                let ptr = peer as *mut $rust_type;
-                drop(Vec::from_raw_parts(ptr, len, len));
+                let raw_slice = Box::from_raw(peer.cast::<*mut [u8]>());
+                let len = raw_slice.len();
+                drop(Vec::from_raw_parts(raw_slice.cast::<u8>(), len, len));
```

Comment:
nit: are we guaranteed to have `u8` here (e.g. maybe it can be a `Vec<i32>` in the macro call)

Reply:
Yes, you are right, this is mistake, I'll fix it.